c/tests/test_trees.c

```diff
@@ -4388,6 +4405,13 @@ test_single_tree_map_mutations(void)
     CU_ASSERT_EQUAL_FATAL(num_transitions, 3);
     free(transitions);
 
+    ancestral_state = 5;
+    ret = tsk_tree_map_mutations(&t, genotypes, NULL, TSK_MM_FIXED_ANCESTRAL_STATE,
+        &ancestral_state, &num_transitions, &transitions);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    CU_ASSERT_EQUAL_FATAL(num_transitions, 4);
+    free(transitions);
```

Comment:
Missing the test for bad_ancestral state now - need to add another check. Also, good to check that ancestral_state is 5 in this one.